### PR TITLE
Sort methods in the property selector alphabetically

### LINKED ISSUE #39576: When inserting function in AnimationPlayer using call method track functions are listed in different order than in a source script
**Godot version:**

3.2.1 

**OS/device including version:**

Win 10 

**Issue description:**

When you use animation player and add new track call method track and then insert key inside this track. Methods listed from a script are placed in seemingly random order rather than in order they appear in the scrip

**Steps to reproduce:**

Described above

**Minimal reproduction project:**

applicable in any project.

### PR DESCRIPTION
**Pull request by Sl3dge78**
Method order seemingly random in PropertySelector when inserting function in AnimationPlayer

Fixes #39576
Method name order is now sorted alphabetically through a custom sorter.

### CODE REVIEW

**Comment by aaronfranke** [comment]
@Sl3dge78 Is this still desired? If so, it needs to be [rebased](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#the-interactive-rebase) and tested on the latest master branch. While there are no conflicts, rebasing is important so that reviewers can easily test the PR.

**Comment by Sl3dge78** [comment]
This was an old quick thing I worked on, I'll to rebase it on current version when I have some free time and do a new PR.

**Comment by akien-mga** [comment]
Sorry for the late review. That solution makes sense to me, something similar is done in `editor/connections_dialog.cpp` too.

I checked and listing the methods in the order they're defined in the script wouldn't be trivial, it's done by `ScriptTextEditor::get_functions` but this requires having access to the script text and its source path, which seems overkill for this.

Let me know if you're able to rebase this PR, otherwise I can have a look myself.

File: editor/property_selector.cpp [hunk 1]
```diff
@@ -203,13 +210,25 @@ void PropertySelector::_update_search() {
 			Object *obj = ObjectDB::get_instance(script);
 			if (Object::cast_to<Script>(obj)) {
 				methods.push_back(MethodInfo("*Script Methods"));
-				Object::cast_to<Script>(obj)->get_script_method_list(&methods);
+
+				List<MethodInfo> obj_methods;
+				Object::cast_to<Script>(obj)->get_script_method_list(&obj_methods);
+				obj_methods.sort_custom<_SortMethodInfoByName>();
```
Comment (TokageItLab) [inline editor/property_selector.cpp:216]: Is .sort() not sufficient?

**Comment by akien-mga** [comment]
Closing as it's abandoned, could be salvaged by anyone interested.

**Pull request closed without merging**